design/blueprint_store.md

```diff
@@ -1,50 +1,227 @@
+            rrb.View3D(
+                root="world",
+                content=[
+                    rrb.Data("./points", visible_history=10).override(radius=0.1, color="blue"),
+                    rrb.Data("./camera").override(image_plane=3),
+                ],
+            ),
+            rrb.HorizontalLayout(
+                content=[
+                    rrb.View2D(
+                        root="world/camera/image",
+                        content=rrb.DataGroup(
+                            rrb.Data("."),
+                            rrb.Points2D("world/points"),
+                        ).default(radius=0.2),
+                    ),
+                    rrb.ViewTimeSeries("metrics", content=rrb.RecursiveData(".")),
+                ],
+            ),
+        ],
+    ),
+)
 
-## Viewer
-The viewer state would be completely driven by the blueprint store and the data store. Each frame we would basically query the blueprint store about the current state of the ui, which would then dictate the queries done to the data store.
+rr.spawn(blueprint)
+```
+
+The assorted objects used in blueprint construction are:
+ - `App`: Container for top-level application state such as panel-visibility, menus, etc.
+ - `TimeControl`: Specific state relevant to the time controls.
+ - `Tile`: Common base-class between `rrb.Container` and `rrb.SpaceView` (follows egui-tile-tree)
+    - `Container`: A layout of components (interface only)
+        - `HorizontalLayout`
+        - `VerticalLayout`
+        - ... additional layouts
+    - `SpaceView`: An actual view of data
+        - `View2D`
+        - `View3D`
+        - `ViewTimeSeries`
+        - ... additional space-views
+ - `DataGroup`: A group of data entities with potentially shared overrides or defaults.
+    - `RecursiveData`: A special DataGroup that recursively includes all data entities under a given path.
+ - `Data`: A singular data entity.
+    - `Points2D`: A typed data entity for 2d points
+    - `Points3D`: A typed data entity for 3d points
+    - `Image`: A typed data entity for images
+    - ... additional typed data entities
+
+Many Blueprint objects will allow for flexible upcasting into an obvious parent-type to reduce unnecessary typing. In
+particular:
+ - `Data` -> `DataGroup`
+ - `[Data]` -> `DataGroup`
+ - `DataGroup` -> `SpaceView` (If view category can be inferred)
+ - `Tile` -> `Viewport`
+ - `Viewport` -> `App`
+
+This means a trivial expression like: `rr.show(rrb.Points3D("points"))` is still a valid Blueprint.
+
+## Blueprint-Static Data
+
+As a further simplification, the Blueprint will allow for the direct inclusion of static data, allowing users to bypass
+the data-logging APIs entirely for simple use-cases that don't require temporal information.  This will be accomplished
+by allowing `rrb.Data` objects to be constructured from any Rerun-loggable object.
+
+Data that is a *query* from the data store references an entity path used seprately by the logging APIs:
```

Comment:
I guess we should call it the "recording store" to be consistent with the decision in todays standup (there are Recording Stores and Blueprint Stores)

```suggestion
Data that is a *query* from the recording store references an entity path used seprately by the logging APIs:
```